### PR TITLE
linear-mixed-effects: expose random slope/intercept parameters

### LINKED ISSUE #65: linear-mixed-effects: expose random slope/intercept parameters

### PR DESCRIPTION
**Pull request by nbokulich**
fixes #65

### CODE REVIEW

File: q2_longitudinal/tests/test_longitudinal.py [hunk 1]
```diff
@@ -284,6 +284,28 @@ def test_linear_mixed_effects(self):
             group_categories='delivery,diet,antiexposedall',
             individual_id_column='studyid', metric='observed_otus')
 
+    def test_linear_mixed_effects_no_group_categories(self):
```
Comment (thermokarst) [inline q2_longitudinal/tests/test_longitudinal.py]: This test isn't asserting the results are correct --- we have some examples of cracking open a viz and testing the contents in a few place (e.g. [q2-demux](https://github.com/qiime2/q2-demux/blob/master/q2_demux/tests/test_demux.py#L757-L783), [q2-diversity](https://github.com/qiime2/q2-diversity/blob/master/q2_diversity/tests/test_alpha_rarefaction.py), [q2-taxa](https://github.com/qiime2/q2-taxa/blob/master/q2_taxa/tests/test_visualizer.py)) --- I think that should be doable here to make sure the grouping working correctly.

File: q2_longitudinal/tests/test_longitudinal.py [hunk 2]
```diff
@@ -284,6 +284,28 @@ def test_linear_mixed_effects(self):
...
+        linear_mixed_effects(
+            output_dir=self.temp_dir.name, table=None,
+            metadata=self.md_ecam_fp, state_column='month',
+            individual_id_column='studyid', metric='observed_otus')
+
+    def test_linear_mixed_effects_with_random_effects(self):
```
Comment (thermokarst) [inline q2_longitudinal/tests/test_longitudinal.py]: You could test for the presence of the new row in the top table of this viz:

```python
self.assertTrue('Random effects</th><td>[month]' in viz)
```

File: q2_longitudinal/tests/test_longitudinal.py [hunk 3]
```diff
@@ -284,6 +284,28 @@ def test_linear_mixed_effects(self):
...
+        linear_mixed_effects(
+            output_dir=self.temp_dir.name, table=None,
+            metadata=self.md_ecam_fp, state_column='month',
+            group_categories='delivery,diet,antiexposedall',
+            random_effects='month',
+            individual_id_column='studyid', metric='observed_otus')
+
+    def test_linear_mixed_effects_with_multiple_random_effects(self):
```
Comment (thermokarst) [inline q2_longitudinal/tests/test_longitudinal.py]: See comment above on `test_linear_mixed_effects_with_random_effects`

File: q2_longitudinal/_utilities.py [hunk 4]
```diff
@@ -381,19 +393,21 @@ def _regplot_subplots_from_dataframe(state_column, metric, metadata,
                                      group_by, lowess=False, ci=95,
                                      palette='Set1'):
     '''plot a single regplot for each group in group_by.'''
+    if group_by is None:
+        # create dummy group column in metadata so we can squeeze into a
```
Comment (thermokarst) [inline q2_longitudinal/_utilities.py:397]: Fair enough!

**Review by thermokarst — CHANGES_REQUESTED** [review]
This looks great @nbokulich, just a few minor requests in the testing department, otherwise this looks good to go! Let me know if you have any questions!

File: q2_longitudinal/tests/test_longitudinal.py (hunk 1 above)
Comment (nbokulich) [reply q2_longitudinal/tests/test_longitudinal.py]: oops... I was being sneaky, wasn't I?

I've added `pdt.assert_frame_equal` comparisons for all LME tests now.

File: q2_longitudinal/tests/test_longitudinal.py (hunk 2 above)
Comment (nbokulich) [reply q2_longitudinal/tests/test_longitudinal.py]: I now do a full frame comparison on the model results, which goes above and beyond this suggestion. Thanks!

File: q2_longitudinal/tests/test_longitudinal.py (hunk 3 above)
Comment (nbokulich) [reply q2_longitudinal/tests/test_longitudinal.py]: 👍

**Comment by nbokulich** [comment]
Thanks @thermokarst ! I have added proper results comparisons for the LME tests

**Review by thermokarst — APPROVED** [review]

**Pull request merged**